docs/DeveloperGuide.md

```diff
@@ -1,38 +1,536 @@
+command interface. The `Unix` parser is designed such that input commands are required to
+have the syntax of ```command -(options) argument1 argument2 ...``` where the order of arguments
+does not matter. While this is a clean and neat parser design, we have decided not to adopt this
+design as we see a strong need of having a hierarchical command structure for our program.
+
+1. A hierarchical structure introduces logical sequence and flow to the command used,
+   facilitating a quick familiarisation process for users to quickly learn the command syntax.
+   This is in line with our design consideration to minimise learning and familiarisation time
+   for users. This is in addition to the ease of processing and validation of commands.
+   ```short [question]/[answer]/[module]/[difficulty]```, for one, is a command that has a
+   logical sequence to which the details of a question is entered. It is more intuitive to
+   follow a structure instead of entering such details without a fixed order, which can easily
+   lead to confusion. <br><br>
+2. The use of the delimiter `/` also better helps to introduce the concept of hierarchy than the
+   delimiter `--` used in `Unix` commands as `/` can be more easily associated with layering and
+   stacking, which create structures possessing hierarchy.
+
+### Parser Command Syntax
+
+Our final design  seeks to optimise both user experience and program effectiveness.
+We have hence chosen the following general syntax for commands to be input into the `Parser`.
+All commands require a starting payload, but some commands do not require the subsequent arguments.
+
+```
+commandType [payload] [/argument1 [payload1] /argument2 [payload2] ... ]
+```
+The command structure is simple, short, and intuitive. With commands rarely requiring more
+than 2 arguments, typing and inputting the command in CLI is fast and efficient, allowing
+revising students to quickly utilise the program. The hierarchical and logical structure to
+the arrangement of arguments and payloads allows students to quickly familiarise with the
+syntax and minimise delays to their learning schedule.
+
+The hierarchical structure also allows for the respective payload to be easily extracted
+with little ambiguity. Command validation can then be performed sequentially, facilitating
+efficient checks.
+
+### Implementation
+
+The main function of `Parser` is to decode user input and extract relevant information to be
+used by other classes. Hence, it supports 3 main methods as listed below.
+1. `parseCommand`
+2. `getContentAfterKeyword`
+3. `extractQuestionDifficulty`
+
+`parseCommand`
+
+This method processes raw user input from CLI to extract information to be assembled into a
+`Command` object. It determines the type of command by reading the commandType from the
+very start of the user input.
+
+The following sequence diagram shows the implementation of `parseCommand`.
+
+![](./UML/Parser.jpg)
```

Comment:
I suggest reducing the amount of commands included in the uml diagram as it will improve readability and still convey the same idea.